echo/echo.py

```diff
@@ -1,131 +1,161 @@
-    # Called when connected
-    def __init__(self, io, path):
-        super().__init__(io, path)
+LOG = logging.getLogger(__name__)
 
-        self.id = None
-        self.emit('ready')
 
-    def on_joined_room(self, data):
-        self.id = data['user']
+class EchoBot:
+    sio = socketio.Client(logger=True)
+    task_id = None
 
-    @staticmethod
-    def get_message_response(success, error=None):
-        if not success:
-            print("Could not send message:", error)
-            sys.exit(2)
-
-        print("message sent successfully")
-
-    def on_text_message(self, data):
-        if not self.id:
-            return
-
-        sender = data['user']['id']
-        if sender == self.id:
-            return
-
-        print("I got a message, let's send it back!:", data)
-
-        message = data['msg']
-        if message.lower() == "hello":
-            message = "World!"
-        if message.lower() == "ping":
-            message = "Pong!"
-
-        if 'room' in data and data['room']:
-            self.emit("text", {'room': data['room'], 'msg': message}, self.get_message_response)
-        else:
-            print("It was actually a private message oO")
-            self.emit("text", {'receiver_id': data['user']['id'], 'msg': message}, self.get_message_response)
-
-    def on_image_message(self, data):
-        sender = data['user']['id']
-        if sender == self.id:
-            return
-
-        print("I got an image, let's send it back!:", data)
-
-        if 'room' in data and data['room']:
-            self.emit("image", {'room': data['room'],
-                                'url': data['url'],
-                                'width': data['width'],
-                                'height': data['height']},
-                      self.get_message_response)
-        else:
-            print("It was actually a private message oO")
-            self.emit("image", {'receiver_id': data['user']['id'],
-                                'url': data['url'],
-                                'width': data['width'],
-                                'height': data['height']},
-                      self.get_message_response)
-
-    def on_new_task_room(self, data):
-        print("new task room", data)
-        if data['task'] == TASK_ID:
-            self.emit("join_room", {'user': self.id, 'room': data['room']})
-
-
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser(description='Run Echo bot')
-
-    if 'TOKEN' in os.environ:
-        token = {'default': os.environ['TOKEN']}
-    else:
-        token = {'required': True}
+    def __init__(self, token, user, host, port):
+        self.token = token
+        self.user = user
 
-    if 'CHAT_HOST' in os.environ:
-        chat_host = {'default': os.environ['CHAT_HOST']}
-    else:
-        chat_host = {'default': 'http://localhost'}
+        self.uri = host
+        if port is not None:
+            self.uri += f":{port}"
+        self.uri += "/slurk/api"
```

Comment:
When applying the suggestion below, the port is guaranteed to be set:
```suggestion
        self.uri = f"{host}:{port}/slurk/api"
```